docs/index.bs

```diff
@@ -3481,93 +3452,79 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
                       1. If |r|'s [=request/url=]'s [=url/scheme=] is not one of "<code>http</code>" and "<code>https</code>", <a>throw</a> a <code>TypeError</code>.
                       1. If |r|'s [=request/method=] is not \`<code>GET</code>\`, <a>throw</a> a <code>TypeError</code>.
                       1. If |operation|.{{CacheBatchOperation/options}} is not null, <a>throw</a> a <code>TypeError</code>.
-                      1. Set |requestResponseArray| to the result of running <a>Query Cache</a> algorithm passing |operation|.{{CacheBatchOperation/request}}.
-                      1. If |requestResponseArray| is not an empty array, then:
-                          1. Let |requestResponse| be |requestResponseArray|[0].
-                          1. Let |fetchingRecord| be the corresponding <a>fetching record</a> for |requestResponse|[0] and |requestResponse|[1] in <a>request to response map</a>.
-                          1. Set |fetchingRecord|.\[[key]] to |operation|.{{CacheBatchOperation/request}} and |fetchingRecord|.\[[value]] to |operation|.{{CacheBatchOperation/response}}.
-                      1. Else:
-                          1. Set a newly-created <a>fetching record</a> {\[[key]]: |operation|.{{CacheBatchOperation/request}}, \[[value]]: |operation|.{{CacheBatchOperation/response}}} to <a>request to response map</a>.
-
-                          Note: The cache commit is allowed as long as the response's headers are available.
-
+                      1. Set |requestResponseList| to the result of running [=Query Cache=] with |operation|.{{CacheBatchOperation/request}}.
+                      1. If |requestResponseList| [=list/is not empty=], [=list/replace=] the [=list/item=] of |cache| that matches |requestResponseList|[0] with |operation|.{{CacheBatchOperation/request}}/|operation|.{{CacheBatchOperation/response}}.
+                      1. Else, [=list/append=] |operation|.{{CacheBatchOperation/request}}/|operation|.{{CacheBatchOperation/response}} to |cache|.
                       1. If the cache write operation in the previous two steps failed due to exceeding the granted quota limit, <a>throw</a> a "{{QuotaExceededError}}" {{DOMException}}.
-                      1. Add an array [|operation|.request, |operation|.response] to |addedRecords|.
-                  1. Add |operation|.response to |resultArray|.
-              1. Return |resultArray|.
+                      1. [=list/Append=] |operation|.{{CacheBatchOperation/request}}/|operation|.{{CacheBatchOperation/response}} to |addedItems|.
+                  1. [=Append=] |operation|.{{CacheBatchOperation/request}}/|operation|.{{CacheBatchOperation/response}} to |resultList|.
+              1. Return |resultList|.
           1. And then, if an exception was <a lt="throw">thrown</a>, then:
-              1. Set the <a>context object</a>'s <a>request to response map</a> to |itemsCopy|.
+              1. Set |cache| to |itemsCopy|.
```

Comment:
Isn't this just setting a local variable? As in, it no longer reverts the operation.

Again, this isn't a new problem, but isn't there a bit of a race condition here? Since we're replacing the whole cache with an older copy, there may be concurrent operations of this resulting in data loss.

Reply:
> Isn't this just setting a local variable?

I think |cache| becomes a reference to the request response list. It's confusing though.

> isn't there a bit of a race condition here?

I wanted making the cache write atomic (the step 3.3) would do some magic, but it's absolutely a part that I should audit and improve.